frontend/src/views/supp-allowances/SupplementarySubmitView.vue

```diff
@@ -228,6 +238,7 @@ export default {
     await this.loadData()
   },
   methods: {
```

Comment:
this issue is not related to this PR. just something I realize while I test this PR. For Impersonate or user with readonly access, I think we should hide the "Cancel" button and we should have a readonly check for saveApplication function (i.e.: we should not trigger saveApplication if it is readonly - in both SupplementaryFormView and SupplementarySubmitView): 
![image](https://github.com/user-attachments/assets/5b6d2107-b92b-4ff3-a537-560f735d4b50)

Reply:
good point I will change that

Reply:
When I click "Back" from the Submit page (Impersonate), I'm still see this error (403 Forbidden for patch request)